pkg/addons/openebs/openebs.go

```diff
@@ -56,63 +57,65 @@ func (o *OpenEBS) HostPreflights() (*v1beta2.HostPreflightSpec, error) {
 	return nil, nil
 }
 
-func (o *OpenEBS) Apply(ctx context.Context) error {
-	loading := pb.Start()
-	loading.Infof("Applying OpenEBS addon")
-	defer loading.Close()
-	version, err := o.latest()
-	if err != nil {
-		return fmt.Errorf("unable to get latest version: %w", err)
-	}
-	if !semver.IsValid(version) {
-		return fmt.Errorf("unable to parse version %s", version)
+func (o *OpenEBS) GetChartFileName() string {
+	return fmt.Sprintf("openebs-%s.tgz", appVersion)
+}
+
+func (o *OpenEBS) GenerateHelmConfig(ctx *cli.Context, isUpgrade bool) ([]dig.Mapping, error) {
+
+	chartConfigs := []dig.Mapping{}
+
+	chartConfig := dig.Mapping{
+		"name":      releaseName,
+		"namespace": o.namespace,
+		"version":   appVersion,
 	}
-	fname := fmt.Sprintf("openebs-%s.tgz", strings.TrimPrefix(version, "v"))
-	hfp, err := charts.FS.Open(fname)
+
+	chartConfig["chartName"] = filepath.Join(defaults.HelmChartSubDir(), o.GetChartFileName())
+
+	valuesStringData, err := yaml.Marshal(helmValues)
 	if err != nil {
-		return fmt.Errorf("unable to find version %s: %w", version, err)
+		return chartConfigs, err
```

Comment:
this should return nil and the error. i suspect you have done this by design so i won't comment on it anymore.